File: 1024.py
def desloca3bytes(str):
    nova = ""
    for c in str:
        if ('A' <= c <= 'Z') or ('a' <= c <= 'z'):
            nova += chr(ord(c) + 3)
        else:
            nova += c

    return nova

def move1byteesquerda(linha):
    tam = len(linha)
    metade = tam // 2

    nova = list(linha)  # transforma string em lista (mutável)
    for j in range(metade, tam):
        nova[j] = chr(ord(nova[j]) - 1)

    return "".join(nova)


n = int(input())
for i in range(n):
    linha = input()

    linha = desloca3bytes(linha)
    linha = linha[::-1]
    linha = move1byteesquerda(linha)

    print(linha)

File: 1024.cpp
#include <algorithm>
#include <iostream>
#include <string>
using namespace std;

void desloca3bytes(string &linha) {
    for(char &c : linha) {
        if ((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z')) {
            c += 3;
        }
    }
}
void revertefrase(string &linha) {
    reverse(linha.begin(), linha.end());
}

void move1byteesquerda(string &linha) {
    int tam = linha.size();
    int metade = tam / 2;

    int j;
    for(j = metade; j < tam; j++)
        linha[j]--;
}

int main() {
    int i, n;
    string linha;

    cin >> n;
    cin.ignore();

    for(i = 0; i < n; i++) {
        getline(cin, linha);

        desloca3bytes(linha);
        revertefrase(linha);
        move1byteesquerda(linha);

        cout << linha << "\n";
    }


    return 0;
}
